plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,331 @@
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    try:
+        if ISO_MODE == "iso9660":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif ISO_MODE == "rr":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif ISO_MODE == "joliet":
+            opened_iso.rm_hard_link(joliet_path=dest_file)
+        elif ISO_MODE == "udf":
+            opened_iso.rm_hard_link(udf_path=dest_file)
+    except Exception as err:
+        msg = "Failed to delete iso file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_rebuild(src_iso, dest_iso, delete_files_list, add_files_list):
+    global ISO_MODE
+
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(src_iso)
+        if iso.has_rock_ridge():
+            ISO_MODE = "rr"
+        elif iso.has_joliet():
+            ISO_MODE = "joliet"
+        elif iso.has_udf():
+            ISO_MODE = "udf"
+
+        if delete_files_list:
+            for item in delete_files_list:
+                ret, msg = iso_delete_file(iso, item)
+                if ret != 0:
+                    MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+        if add_files_list:
+            for item in add_files_list:
+                ret, msg = iso_add_file(iso, item['src_file'], item['dest_file'])
+                if ret != 0:
+                    MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+        iso.write(dest_iso)
+    except Exception as err:
+        msg = "Failed to rebuild ISO %s with error: %s" % (src_iso, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+    finally:
+        if iso:
+            iso.close()
+
+
+def main():
+    op_data_list = []
+    global MODULE_ISO_CUSTOMIZE
+
+    argument_spec = dict(
+        src_iso=dict(type='str', required=True),
+        dest_iso=dict(type='str', required=True),
+        delete_files=dict(type='list', required=False, elements='str'),
+        add_files=dict(type='list', required=False, elements='dict'),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False,
+    )
+    if not HAS_PYCDLIB:
+        module.fail_json(
+            missing_required_lib('pycdlib'), exception=PYCDLIB_IMP_ERR)
+
+    src_iso = module.params.get('src_iso')
+    if not os.path.exists(src_iso):
+        module.fail_json(msg="The %s does not exist." % src_iso)
+
+    dest_iso = module.params.get('dest_iso')
+    if not dest_iso:
+        msg = 'Please specify the path of the customized ISO file using dest_iso parameter.'
+        module.fail_json(str(msg=msg))
+
+    dest_iso_dir = os.path.dirname(dest_iso)
+    if dest_iso_dir and not os.path.exists(dest_iso_dir):
+        module.fail_json(msg="The dest directory %s does not exist" % dest_iso_dir)
+
+    MODULE_ISO_CUSTOMIZE = module
+
+    delete_files_list = module.params.get('delete_files')
+    add_files_list = module.params.get('add_files')
```

Comment:
Suggestion: strip the strings here
```suggestion
    delete_files_list = [s.strip() for s in module.params.get('delete_files')]
    add_files_list = [s.strip() for s in module.params.get('add_files')]
```
And then remove the `.strip()` in a dozen different places in the code.

Reply:
Fix it.
Won't change the line "add_files_list = module.params.get('add_files')" because It return a dict for module.params.get('add_files').